public/pages/CreateSnapshotPolicy/components/CronSchedule/CronSchedule.tsx

```diff
@@ -0,0 +1,173 @@
+  EuiLink,
+  EuiSelect,
+  EuiSpacer,
+  EuiText,
+} from "@elastic/eui";
+import CustomLabel from "../../../../components/CustomLabel";
+import { WEEK_DAYS, CRON_SCHEDULE_FREQUENCY_TYPE, TIMEZONES } from "./constants";
+import { buildCronExpression, parseCronExpression, startTime } from "./helper";
+import moment from "moment-timezone";
+import { CRON_EXPRESSION_DOCUMENTATION_URL } from "../../../../utils/constants";
+
+interface CronScheduleProps {
+  frequencyType: string;
+  onChangeFrequencyType: (e: ChangeEvent<HTMLSelectElement>) => void;
+  cronExpression: string;
+  onCronExpressionChange: (expression: string) => void;
+  timezone?: string;
+  onChangeTimezone?: (e: ChangeEvent<HTMLSelectElement>) => void;
+}
+
+const CronSchedule = ({
+  frequencyType,
+  onChangeFrequencyType,
+  cronExpression,
+  onCronExpressionChange,
+  timezone,
+  onChangeTimezone,
+}: CronScheduleProps) => {
+  const { minute: initMin, hour: initHour, dayOfWeek: initWeek, dayOfMonth: initMonth } = parseCronExpression(cronExpression);
+
+  const [minute, setMinute] = useState(initMin);
+  const [hour, setHour] = useState(initHour);
+  const [dayOfWeek, setWeek] = useState(initWeek);
+  const [dayOfMonth, setMonth] = useState(initMonth);
+
+  useEffect(() => {
+    changeCron();
+  }, [minute, hour, dayOfWeek, dayOfMonth]);
+
+  const changeCron = (input?: any) => {
+    let cronParts = { hour, minute, dayOfWeek, dayOfMonth, frequencyType };
+    cronParts = { ...cronParts, ...input };
+    const expression = buildCronExpression(cronParts, cronExpression);
+    console.log(`sm dev built expression ${expression}`);
+    onCronExpressionChange(expression);
+  };
+
+  function onDayOfWeekChange(dayOfWeek: string) {
+    setWeek(dayOfWeek);
+    // changeCron({ dayOfWeek });
+  }
+
+  function onDayOfMonthChange(dayOfMonth: number) {
+    setMonth(dayOfMonth);
+    // changeCron({ dayOfMonth });
+  }
+
+  function onStartTimeChange(date: moment.Moment) {
+    const minute = date.minute();
+    const hour = date.hour();
+    setMinute(minute);
+    setHour(hour);
+    // changeCron({ minute, hour });
+  }
+
+  function onTypeChange(e: ChangeEvent<HTMLSelectElement>) {
+    const frequencyType = e.target.value;
+    onChangeFrequencyType(e);
+    changeCron({ frequencyType });
+  }
+
+  const dayOfWeekCheckbox = (day: string, checkedDay: string) => (
+    <EuiFlexItem key={day} grow={false} style={{ marginRight: "0px" }}>
+      <EuiCheckbox id={day} label={day} checked={checkedDay === day} onChange={(e) => onDayOfWeekChange(day)} compressed />
+    </EuiFlexItem>
+  );
+
+  let additionalContent;
+  if (frequencyType === "weekly") {
+    additionalContent = <EuiFlexGroup>{WEEK_DAYS.map((d) => dayOfWeekCheckbox(d, dayOfWeek))}</EuiFlexGroup>;
+  }
+  if (frequencyType === "monthly") {
+    additionalContent = (
+      <EuiFlexGroup gutterSize="m">
+        <EuiFlexItem>
+          <EuiSelect options={[{ value: "day", text: "Day" }]} defaultValue="Day" />
```

Comment:
I haven't seen the mocks for this feature, but some help text here might be useful. It's not immediately obvious what `Day` followed by a number means in my opinion.
![Screen Shot 2022-06-21 at 5 11 32 PM](https://user-images.githubusercontent.com/79280347/174916884-39ecfb76-04d1-4900-b4a0-065f1c1c1a6f.png)

Reply:
Good point, add "On the" title above it.